R/html_dependencies.R

```diff
@@ -539,3 +539,15 @@ html_dependency_tabset <- function() {
     all_files = FALSE
   )
 }
+
+#' @rdname html-dependencies
+#' @export
+html_dependency_codefolding_lua <- function() {
+  htmlDependency(
+    "codefolding-lua",
+    version = "1.1",
+    src = pkg_file("rmd/h/navigation-1.1"),
+    stylesheet = "codefolding-lua.css",
+    all_files = FALSE
+  )
+}
```

Comment:
> having a css file in addition to the lua files is too much

I agree. I remember that is part of the reason why I introduced `tabset.js` instead of `tabset.lua`.
Maybe code folding should also be implemented in JavaScript.
If we will implement Hide/Show All buttons, then we will need JavaScript anyway...

Answering your suggestion, I would not include CSS by Lua filter because different formats would require different styling.

For example, if we implement by JS and CSS, then we can easily opt-out CSS.
If we do the similar thing by Lua filter, we need to pass the argument to Lua filter via Pandoc variable or metadata.

``` r
html_dependency_codefolding <- function(stylesheet = TRUE) {
  htmlDependency(
    "codefolding-lua",
    version = "1.1",
    src = pkg_file("rmd/h/navigation-1.1"),
    script = "codefolding.js",
    stylesheet = if (stylesheet) "codefolding.css",
    all_files = FALSE
  )
}
```

Reply:
> Answering your suggestion, I would not include CSS by Lua filter because different formats would require different styling.

You're right. That is a good argument. Thanks!

> If we will implement Hide/Show All buttons, then we will need JavaScript anyway...

You could include the JS part for this as part of the Lua filter 😝 ... kidding 

> Maybe code folding should also be implemented in JavaScript.

I like the fact that this is done in Lua because we are touching the structure of the HTML and not just changing attributes or something else. 

Let's leave it like that then, as this is how we do it for now and this is working well. 

More general thought: 

I think I have a wish in mind I should start working on at some point: a way to have a _feature extensions bundle_ where we have a  `htmlDependency` with a Lua filter requirements. Something that we could write in a wrapped way, and that **rmarkdown** could unwrapped and use the correct way. We don't have a `lua_dependency` component, and maybe that would make sense to have at some point adding more feature like this 
````r
lua_dependency(
	filters = "code-folding-lua",
    html_deps = htmlDependency(
    	"codefolding-lua",
	    version = "1.1",
    	src = pkg_file("rmd/h/navigation-1.1"),
	    stylesheet = "codefolding-lua.css",
    	all_files = FALSE
  )
}
````

We could then group some feature. Just an idea...

Reply:
`html_dependency` with Lua filter!! That is what I exactly want!!

I would rather name it like `rmd_dependency` or some other more generic name so that users only need to use it instead of using `html_dependency` and `lua_dependency`.

Reply:
Glad you think it is a good idea. I'll open an issue to track the idea.